WordPress/Sniffs/Theme/NoTitleTagSniff.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of the <title> tag, unless it is within a <svg> tag.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NoTitleTagSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Property to keep track of whether a <svg> open tag has been encountered.
+	 *
+	 * @var bool
+	 */
+	private $in_svg = false;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_INLINE_HTML, // Finds inline html.
+			T_CONSTANT_ENCAPSED_STRING, // Finds the title tag in PHP.
+		 );
+	} // end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token in
+	 *                                        the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+		$tokens  = $phpcsFile->getTokens();
+		$content = $tokens[ $stackPtr ]['content'];
+
+		// No need to check an empty string.
+		if ( '' === trim( $content ) ) {
+			return;
+		}
+
+		// Are we in a <svg> tag ?
+		if ( true === $this->in_svg ) {
+			if ( false === strpos( $content, '</svg>' ) ) {
+				// Rest in_svg at the end of the file.
+				if ( ( $phpcsFile->numTokens - 1 ) === $stackPtr ) {
```

Comment:
This will only work if the last token in the file is one of the two type of tokens you are listening for.
We need a better solution for this.
While it makes things more complicated, we may need to keep track of `in_svg` per sniffed file like
```php
$this->in_svg[ $filename ] = true/false;
```

That way at least there won't be any cross-file contamination of the property.